src/sentry/api/endpoints/auth_merge_user_accounts.py

```diff
@@ -0,0 +1,40 @@
+from django.contrib.auth.models import AnonymousUser
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.paginator import OffsetPaginator
+from sentry.api.permissions import SentryIsAuthenticated
+from sentry.api.serializers import serialize
+from sentry.users.api.serializers.user import UserSerializerWithOrgMemberships
+from sentry.users.models.user import User
+
+
+@control_silo_endpoint
+class AuthMergeUserAccountsEndpoint(Endpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    permission_classes = (SentryIsAuthenticated,)
+    """
+    List and merge user accounts with the same primary email address.
+    """
+
+    def get(self, request: Request) -> Response:
+        user = request.user
+        if isinstance(user, AnonymousUser):
+            return Response(
+                status=401,
+                data={"error": "You must be authenticated to use this endpoint"},
+            )
+
+        shared_email = user.email
```

Comment:
should we assert that this is not blank? if someone logged in with a username + password combo could the email be empty? or somebody just hits the API directly (even though it's undocumented)